### PR TITLE
Enable Handling Error

### PR DESCRIPTION
**Pull request by mosson**
I want to test of handling error.
Please see if this is mergeable..
Thanks!

### CODE REVIEW

**Comment by oziks** [comment]
Thank you for your contribution.

Please, update the readme [usage section](https://github.com/M6Web/superagent-mock#usage) by adding the `match`parameter of the `fixtures` callback + doc comment.

**Comment by oziks** [comment]
Could you squashing your commits please ? :smiling_imp:

**Comment by mosson** [comment]
I squashed. 
Please see if this is mergeable..
Thanks!

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+.idea
```
Comment (fdubost) [inline .gitignore]: You should add this in your global gitignore, it's specific to your configuration :wink:

**Comment by fdubost** [comment]
I don't understand why you don't throw the error in the callback function, you will get the same behaviour, no :question: For me, it's unnecessary to pass matching data to the fixtures function.

**Comment by mosson** [comment]
@fdubost Thanks for reviewing.

I'm trying to explain below that You point 2 features out. 

At first, Why throw the error in the callback function.
I think that throwing errors in the callbacks and configuring to throw error in spec are different.
For example, 

```
// assuming production code.

require('superagent')
  .get('/foo/bar')
  .end(function(err, res){
    if (err) {
      alert('something went to wrong');
      // our clients receive a message that an error occurs.
      // I need to test to invoke this line.
    }
  });
```

If throws an error in callback, anyone catches it, or add some line to catch it in production code.
And It seems to me that above code can not test with `superagent-mock`.
So I send this PR to you. Or do you have any Idea to test above code? I need it.

Second, Why pass matching data to the fixtures function.

> it's unnecessary to pass matching data to the fixtures function.

That's true, but it's convenience of returning dynamic value..

Assuming that
when request get '/foo', response the file content named "foo.json"
when request get '/bar', response the file content named "bar.json" 

Now You need two configuration objects to make this behavior. But I think It's too long.
if passing matching data to fixture function, It become possible that resolves the file name and the configuration code is now shorter than before.

**Comment by mosson** [comment]
I removed `.idea` from .gitignore and squashed.
Please see if this is mergeable..
Thanks!

**Comment by oziks** [comment]
Thanks for contributing.

**Comment by oziks** [comment]
Project release to v1.1.0 and publish on npmjs.

**Pull request merged**